x/wasm/types/types.go

```diff
@@ -260,17 +260,15 @@ func NewEnv(ctx sdk.Context, contractAddr sdk.AccAddress) wasmvmtypes.Env {
 	if ctx.BlockHeight() < 0 {
 		panic("Block height must never be negative")
 	}
-	sec := ctx.BlockTime().Unix()
-	if sec < 0 {
-		panic("Block (unix) time must never be negative ")
+	nano := ctx.BlockTime().UnixNano()
+	if nano < 1 {
```

Comment:
A 0 time is well supported in this case. It is serialized as `"0"` because of `string` in
```
Time    uint64 `json:"time,string"`
```
. But this way works as well.

Reply:
IMHO there is no realistic scenario with timestamp 0 and we should fail fast. While genesis block has height 0 it does contain a >0 timestamp.